### ISSUE TITLE
Consumer association not checked in Admin API

### ISSUE THREAD

**Issue by thibaultCha** [Mashape/kong#735]
Entities such as `key-auth` credentials can be accessed through, for ex:

```
/consumers/{consumer}/key-auth/{key_id}
```

When giving a `{consumer}` that exists *but* that is not associated with the given `{key_id}`, the `{key_id}` can be updated/deleted, etc... I'm not sure if the way the DB is queried takes the relation into account or simply uses the `{key_id}`.

**Comment by thibaultCha** [comment]
Same with other endpoints, notrelated to consumers. For example:

```
apis/{api}/plugins/{id}
```

is actually equivalent to

```
/plugins/{id}
```

in the sense that it won't even check if this plugin is correctly attached to the given `{api}`.

**Comment by thefosk** [comment]
This issue seems to have been already fixed in the past, as #1071 demonstrates.
